subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -64,17 +64,48 @@ interface Identity {
      */
     WorkOutput execute(ExecutionRequest executionRequest);
 
+    /**
+     * Parameter object for {@link #execute(ExecutionRequest)}.
+     */
     interface ExecutionRequest {
+        /**
+         * The directory to produce outputs into.
+         * <p>
+         * Note: it's {@code null} for tasks as they don't currently have their own workspace.
+         */
         File getWorkspace();
 
+        /**
+         * For work capable of incremental execution this is the object to query per-property changes through;
+         * {@link Optional#empty()} for non-incremental-capable work.
+         * <p>
+         * Note that incremental-capable work can also be executed non-incrementally, but even then a
+         * {@link org.gradle.work.InputChanges} instance is avaialble, returning all inputs as changed.
```

Comment:
```suggestion
         * {@link org.gradle.work.InputChanges} instance is avaialble, returning all inputs as added.
```

Aren't all the inputs returned as added and not as changed? I am also not sure how much this matters here.

Reply:
Good point, removed this clause.